backend/fms_core/management/commands/_delete_individual.py

```diff
@@ -0,0 +1,86 @@
+ACTION = "action"                     # = delete_individual
+CURATION_INDEX = "curation_index"     # Number indicating the order in which this action was performed during the curation.
+COMMENT = "comment"                   # An optional comment to be stored in the logs
+INDIVIDUAL_NAMES = "individual_names" # An array of individual name that are to be deleted.
+USER_ID = "requester_user_id"         # The user id of the person requesting the curation. Optional. If left empty, uses biobankadmin id.
+
+# Curation params template
+# { CURATION_INDEX: 1,
+#   ACTION: "delete_individual",
+#   COMMENT: "Dr. No asked to delete these individuals that are not used.",
+#   INDIVIDUAL_NAMES: ["Roger", "TiPaul", "Gertrude"], # List of individual names (unique) to be deleted
+#   USER_ID: 5
+# }
+
+# This curation deletes only individual that are not referenced. Remove individual from samples references using "update_field_value" first.
+# To delete parents, a first curation need to be run to delete the child individual (or an update), then parents will be unreferenced.
+
+# function that checks the references to an individual and list them.
+def references(individual):
+    links = []
+    children = list(individual.mother_of.all())
+    if children:
+        links.append(f"Mother of {[child.name for child in children]}.")
+        children = None
+    children = list(individual.father_of.all())
+    if children:
+        links.append(f"Father of {[child.name for child in children]}.")
+        children = None
+    samples = list(individual.samples.all())
+    if samples:
+        links.append(f"Has samples {[sample.id for sample in samples]}")
+    return links
+
+
+def delete_individual(params, objects_to_delete, log):
+    log.info("Action [" + str(params[CURATION_INDEX]) + "] Update Field Value started.")
+    log.info("Comment [" + str(params.get(COMMENT, "None")) + "].")
+    log.info("Individual names : " + str(params[INDIVIDUAL_NAMES]) + ".")
+    log.info("Requester id : " + str(params.get(USER_ID)))
+
+    # initialize the curation
+    curation_code = params.get(CURATION_INDEX, "Invalid index")
+    error_found = False
+    name_array = params[INDIVIDUAL_NAMES]
+    user_id = params.get(USER_ID)
+
+    try:
+        individual_model = apps.get_model("fms_core", "Individual")
+        count_deletes = 0
+        for name in name_array:
+            try:
+                individual = individual_model.objects.get(name=name)
+                links = references(individual)
+                if links:
+                    log.error(f"Individual [{name}] is still referenced. {links}")
+                    error_found = True
+                else:
+                    log.info(f"Deleted [Individual] name [{individual.name}] id [{individual.id}]].")
+                    individual.deleted = True
+                    if user_id:
+                        individual.save(requester_id=user_id) # save using the id of the requester
+                    else:
```

Comment:
This logic of verifying whether there is a requester user_id or using the default admin, should not be part of the curation, it should be in the model or TrackedModel. I have seen that there is something there that handles that (might need few improvements)

Reply:
Nah this is part of the curation and the tracked model code supporting this was added for the curation scripts. The model do not care who did what ...

Reply:
I mean the if user_id individual.save(... user_id) else  individual.save()

Reply:
With your current logic by passing an argument, you should handle the general cases where there is no requester_id provided, in the TrackedModel